File: backend/config/src/main/java/org/example/config/EnumConfig.java
package org.example.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;
import org.springframework.core.convert.converter.ConverterRegistry;

@Configuration
public class EnumConfig {
    @Bean
    @Primary
    public ConverterRegistry initConverter(ConverterRegistry registry) {
        registry.addConverterFactory(new EnumConverter());
        return registry;
    }
}

File: laptop-store-backend/config/src/main/kotlin/org/example/config/EnumConfig.kt
package org.example.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.core.convert.converter.ConverterRegistry

@Configuration
class EnumConfig {
    @Bean @Primary
    fun initConverter(registry: ConverterRegistry): ConverterRegistry {
        registry.addConverterFactory(EnumConverter())
        return registry
    }
}